addons/web_editor/static/src/js/wysiwyg/widgets/color_palette.js

```diff
@@ -0,0 +1,320 @@
+            this.$('button.selected').removeClass('selected');
+            $(target).addClass('selected');
+        }
+        this.trigger_up('color_picked', {
+            cssColor: target.style.backgroundColor || this.options.colorPrefix + target.dataset.color,
+            isClass: !!target.dataset.color,
+            target: ev.target,
+        });
+    },
+    /**
+     * Called when a color button is entered.
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _onColorButtonEnter: function (ev) {
+        ev.stopPropagation();
+        const target = ev.currentTarget;
+        this.trigger_up('color_hover', {
+            cssColor: target.style.backgroundColor || this.options.colorPrefix + target.dataset.color,
+            isClass: !!target.dataset.color,
+            target: ev.target,
+        });
+    },
+    /**
+     * Called when a color button is left.
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _onColorButtonLeave: function (ev) {
+        ev.stopPropagation();
+        let params = {
+            cssColor: '',
+            isClass: false,
+            target: ev.target,
+        };
+        if (this.options.$target) {
+            const selected = this.el.querySelector('button.selected');
+            if (selected) {
+                params = {
+                    cssColor: selected.style.backgroundColor || this.options.colorPrefix + selected.dataset.color,
+                    isClass: !!selected.dataset.color,
+                    target: ev.target,
+                };
+            }
+        }
+        this.trigger_up('color_leave', params);
+    },
+    /**
+     * Called when the color reset button is clicked.
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _onColorResetButtonClick: function (ev) {
+        this.$('button.selected').removeClass('selected');
+        this.trigger_up('color_reset', {
+            target: ev.target,
+        });
+    },
+    /**
+     * Called when the custom color button is clicked.
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _onCustomColorButtonClick: function (ev) {
+        ev.stopPropagation();
+        const originalEvent = ev;
+        const $selected = this.$('button.selected')
+        const colorpicker = new ColorPickerDialog(this, {
+            defaultColor: this.options.defaultColor || $selected.css('background-color'),
+        });
+        colorpicker.on('colorpicker:saved', this, ev => {
+            $selected.removeClass('selected');
+            this._addCustomColorButton(ev.data.cssColor, this.options.$target ? ['selected'] : undefined);
+            this.trigger_up('custom_color_picked', {
+                cssColor: ev.data.cssColor,
+                target: originalEvent.target,
+            })
+        });
+        colorpicker.open();
+    },
+};
+
+let ColorPaletteWidget = Widget.extend(ColorPaletteMixin);
+
+let ColorPaletteDialog = Dialog.extend(ColorPaletteMixin, {
```

Comment:
Is the dialog still useful? I think we won't use it anymore? Always an inline palette which could open the custom colorpicker dialog?

In any case, I think that if we need it it should be a Dialog which instantiates a ColorPaletteWidget rather that both using a mixin.